[test] #2272: Add tests for 'FindAssetDefinitionById' query

pesterev opened the pull request:
<!-- You will not see HTML commented line in Pull Request body -->
<!-- Optional sections may be omitted. Just remove them or write None -->

<!-- ### Requirements -->
<!-- * Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion. -->
<!-- * All new code must have code coverage above 70% (https://docs.codecov.io/docs/about-code-coverage). -->
<!-- * CircleCI builds must be passed. -->
<!-- * Critical and blocker issues reported by Sorabot must be fixed. -->
<!-- * Branch must be rebased onto base branch (https://soramitsu.atlassian.net/wiki/spaces/IS/pages/11173889/Rebase+and+merge+guide). -->


### Description of the Change

Add some tests for 'FindAssetDefinitionById' query.

### Issue

Resolves #2272 

### Benefits

It tests the query.

### Possible Drawbacks

None

<!--
NOTE: User may want skip pull request and push workflows with [skip ci]
https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/
Phrases: [skip ci], [ci skip], [no ci], [skip actions], or [actions skip]
-->

s8sato commented on client/tests/integration/queries/asset_definition.rs:
```diff
@@ -0,0 +1,33 @@
+    let asset_definition_id =
+        <AssetDefinition as Identifiable>::Id::from_str("test_coin#wonderland").expect("Valid");
+    let asset_definition = AssetDefinition::quantity(asset_definition_id.clone());
+    test_client.submit_blocking(RegisterBox::new(asset_definition.clone()))?;
+    test_client.submit_blocking(MintBox::new(
```
Here is also worth querying and checking the result

```suggestion
    test_client.submit_blocking(RegisterBox::new(asset_definition.clone()))?;
    // Here
    test_client.submit_blocking(MintBox::new(
```

appetrosyan commented on client/tests/integration/queries/asset_definition.rs:
```diff
@@ -0,0 +1,41 @@
+use test_network::Peer as TestPeer;
+
+#[test]
+fn find_asset_definition_by_id() -> Result<()> {
```
Since this is not what I asked you to do, I would like to know why you decided not to add the query to an existing test.

pesterev replied:
OK, now I got you. So I'm going to move this test into the closest module ([account.rs](https://github.com/hyperledger/iroha/blob/iroha2-dev/client/tests/integration/queries/account.rs)) like this:
```rs
    // Registering accounts
    let register_accounts = accounts
        .iter()
        .skip(1) // Alice has already been registered in genesis
        .cloned()
        .map(|account_id| RegisterBox::new(Account::new(account_id, [])).into())
        .collect::<Vec<_>>();
    test_client.submit_all_blocking(register_accounts)?;

   // HERE
===>  let received_asset_definition = test_client.request(client::asset::definition_by_id(definition_id))?;
===>  assert_eq!(received_asset_definition, asset_definition.build());

    let mint_asset = accounts
        .iter()
        .cloned()
        .map(|account_id| <Asset as Identifiable>::Id::new(definition_id.clone(), account_id))
        .map(|asset_id| MintBox::new(1_u32, asset_id).into())
        .collect::<Vec<_>>();
    test_client.submit_all_blocking(mint_asset)?;

   // HERE AFTER MINTING
===>   let received_asset_definition = test_client.request(client::asset::definition_by_id(definition_id))?;
===>   assert_eq!(received_asset_definition, asset_definition.build());

    let accounts = HashSet::from(accounts);
```

s8sato replied:
I now realize that this PR only needs to replace `FindAllAssetsDefinitions` with `FindAssetDefinitionById` in [this Torii unit test](https://github.com/hyperledger/iroha/blob/1f1c8d79524b4af3536f3505d103014d86304ec9/cli/src/torii/tests.rs#L414-L438)

pesterev replied:
Yeah seems this test needs to replace but its only on the server-side. I also added a new client-side method so I think it also should be tested.

pesterev commented:
@appetrosyan I've removed `asset_definition.rs` module and moved the test into an existing test in `account.rs` module.

@s8sato I've replaced `FindAllAssetsDefinitions` with `FindAssetDefinitionById` in `find_asset_definition`.
